packages/typespec-python/scripts/eng/regenerate.ts

```diff
@@ -371,16 +375,33 @@ function addOptions(spec: string, generatedFolder: string, flags: RegenerateFlag
             options["debug"] = "true";
         }
         options["examples-dir"] = toPosix(join(dirname(spec), "examples"));
-        const configs = Object.entries(options).flatMap(([k, v]) => {
-            return `--option @azure-tools/typespec-python.${k}=${typeof v === "string" && v.indexOf(" ") > -1 ? `"${v}"` : v}`;
-        });
         emitterConfigs.push({
-            optionsStr: configs.join(" "),
+            optionsStr: getConfigsFromOptions(options).join(" "),
             outputDir: options["emitter-output-dir"],
         });
+        if (generateSubdir(spec)) {
+            const packageName = `${options["package-name"] || defaultPackageName(spec)}-with-subdir`;
+            const subdirOptions = {
+                ...options,
+                "package-name": packageName,
+                "generation-subdir": "_generated/",
+                "emitter-output-dir": toPosix(`${generatedFolder}/test/${flags.flavor}/generated/${packageName}/subdir`)
+            }
+            emitterConfigs.push({
+                optionsStr: getConfigsFromOptions(subdirOptions).join(" "),
+                outputDir: subdirOptions["emitter-output-dir"],
+            });
+        }
     }
     return emitterConfigs;
 }
+
+function getConfigsFromOptions(options: Record<string, string>): string[] {
```

Comment:
Additional logic for a special case brings more complexity for regeneration. I think you could just add additional config at https://github.com/Azure/autorest.python/blob/1069e394ec7f54150b83928ea2e2e14da5f21a62/packages/typespec-python/scripts/eng/regenerate.ts#L187-L202

like:

```json
{
        {
            "package-name": "generationsubdir",
            "namespace": "generationsubdir",
            "generation-subdir": "_generated/"
        },
}
```

Reply:
oh I didn't know we had that! yes that would be great, let me update my regenerate.ts